shell/browser/native_browser_view.cc

```diff
@@ -11,8 +11,12 @@ namespace electron {
 
 NativeBrowserView::NativeBrowserView(
     InspectableWebContents* inspectable_web_contents)
-    : inspectable_web_contents_(inspectable_web_contents) {
-  Observe(inspectable_web_contents_->GetWebContents());
+    : content::WebContentsUserData<NativeBrowserView>(
+          *inspectable_web_contents->GetWebContents()),
+      inspectable_web_contents_(inspectable_web_contents) {
+  Observe(inspectable_web_contents->GetWebContents());
+  inspectable_web_contents->GetWebContents()->SetUserData(
+      UserDataKey(), base::WrapUnique(this));
```

Comment:
@deepak1556 @codebytere Making `NativeBrowserView` a subclass of `WebContentsUserData` means the `NativeBrowserView`'s lifetime will be managed by `WebContents`, so with this line the pointer to `NativeBrowserView` is now managed by 2 separate `unique_ptr` in `electron::api::BrowserView` and `WebContents`.

Reply:
@zcbenz ah - unless i add this line though the call to `FromWebContents` is always nullptr. Do you know what I might be missing?

Reply:
The design of WebContentsUserData does not expect users to wrap an existing pointer, users should call CreateForWebContents to create the instance instead.

I think we should not make NativeBrowserView a subclass of WebContentsUserData at all, it is not supposed to be used in this way.

Reply:
@zcbenz updated to take my initial approach!

Reply:
Sorry I was suggesting more of moving the lifetime of `NativeBrowserView` to be owned by `WebContents` since I saw that the view operates on a single `WebContent` for its lifetime. I am good with not changing the lifetime here if that is not the intended contract.